02-microarray/pathway-analysis_microarray_03_gsea.Rmd

```diff
@@ -0,0 +1,493 @@
+
+Let's see a preview of `dge_mapped_df`.
+
+```{r}
+head(dge_mapped_df)
+```
+
+Now let's check to see if we have any Entrez IDs that mapped to multiple Ensembl IDs.
+
+```{r}
+any(duplicated(dge_mapped_df$entrez_id))
+```
+
+Looks like we do have duplicated Entrez IDs.
+Let's find out which Entrez IDs have been duplicated.
+
+```{r}
+dge_mapped_df %>%
+  dplyr::filter(duplicated(entrez_id)) %>%
+  dplyr::pull(entrez_id)
+```
+
+Now let's take a look at the rows associated with one of the duplicated Entrez IDs.
+
+```{r}
+dge_mapped_df %>%
+  dplyr::filter(entrez_id == "336702")
+```
+
+We can see that the associated values vary for each row although the Entrez ID is the same.
+
+We do not want duplicated gene identifiers for the GSEA steps later, so let's keep the Entrez IDs associated with the higher t-statistic value.
+
+```{r}
+filtered_dge_mapped_df <- dge_mapped_df %>%
+  # Sort so that highest t-statistic values are at the top
+  dplyr::arrange(dplyr::desc(t)) %>%
+  # Filter out the duplicated rows using `dplyr::distinct()`-- this will keep
+  # the first row with the duplicated value thus keeping the row with the
+  # highest t-statistic value
+  dplyr::distinct(entrez_id, .keep_all = TRUE)
+```
+
+Let's check to see that we removed the duplicate Entrez IDs and kept the row with the higher t-statistic value.
+
+```{r}
+filtered_dge_mapped_df %>%
+  dplyr::filter(entrez_id == "336702")
+```
+
+Looks like we were able to successfully get rid of the duplicate gene identifiers and keep the observation with the higher t-statistic value!
+Note however, that a caveat in using this approach is that the genes that have duplicate identifiers could be enriched in a particular pathway/gene set and we may get an overly optimistic view of how perturbed that pathway truly is.
+
+**START REVIEW HERE**
+
+## Perform gene set enrichment analysis (GSEA)
+
+The goal of GSEA is to determine whether the members of a set of genes are randomly distributed throughout the pre-ranked gene list or primarily found at the top or bottom of the list.
+Genes are ranked based on phenotype so we expect that sets related to a distinct phenotype will tend to show the latter distribution [@clusterProfiler-book].
+
+### Determine our pre-ranked genes list
+
+The `GSEA()` function takes a pre-ranked (sorted) named vector of statistics, where the names in the vector are gene identifiers.
+This is step 1 -- gene-level statistics.
+
+```{r}
+# Let's create a named vector ranked based on the t-statistic values -- we
+# already sorted in the t-stastic values in descending order in the previous
+# chunk so we won't need to do that here
+t_vector <- filtered_dge_mapped_df$t
+names(t_vector) <- filtered_dge_mapped_df$entrez_id
+```
+
+Let's preview our pre-ranked named vector.
+
+```{r}
+# Look at first entries of the ranked t-statistic vector
+head(t_vector)
+```
+
+### Run GSEA using the `GSEA()` function
+
```

Comment:
Per @jaclyn-taroni's [comment on PR #345](https://github.com/AlexsLemonade/refinebio-examples/pull/345#discussion_r518109401), we will likely want to  move the explanation below up before the `gene ID conversion` section and reiterate here since we now need to explain some scientific decision making in the `gene ID conversion` section.

Reply:
I think we do want to move this up, possibly before we install the libraries or under `Purpose of this analysis`. I would probably collect some of the information from throughout what is being added here (this section, around line 280, around line 395) and write up something pretty comprehensive as an introduction to the method to start. Then I'd _consider_ paring back what is written alongside the code chunks to reiterate the most salient points.